### PR TITLE
Rewrite the automerge stuff in terms of GH Actions

### PR DESCRIPTION
**Pull request by djspiewak**
The current automerge stuff appears to be merging **breaking** PRs, and is also written in terms of Ruby. This is unnecessarily annoying. This PR rewrites it to use the GH Actions conditionals, removing the last need for `octokit`.

### CODE REVIEW

File: core/src/main/scala/precog/SbtPrecogBase.scala [hunk 1]
```diff
@@ -257,20 +250,24 @@ abstract class SbtPrecogBase extends AutoPlugin {
...
             name = Some("Common sbt setup")),
-          WorkflowStep.Run(List("./scripts/commonSetup")),
-          WorkflowStep.ComputeVar("CLONE_DIR", "mktemp -d /tmp/precog-bump.XXXXXXXX"),
-          WorkflowStep.ComputeVar("PR_NUMBER", s"echo $$GITHUB_REF | cut -d'/' -f3"),
-          WorkflowStep.Run(List("./scripts/checkAndAutoMerge"))),
-        cond = Some("github.event_name == 'pull_request' && contains(github.head_ref, 'version-bump')"),
+
+          WorkflowStep.Run(
+            List(
+              "curl -L https://github.com/precog/devtools/raw/master/bin/sdmerge > /tmp/sdmerge",
```
Comment (nicflores) [inline core/src/main/scala/precog/SbtPrecogBase.scala:259]: @djspiewak are we ok with not fixing the version of `sdmerge` for reproducibility here?
I guess `sdmerge` doesn't change very often.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -222,7 +217,7 @@ jobs:
   auto-merge:
     name: Auto Merge
     needs: [build]
-    if: github.event_name == 'pull_request' && contains(github.head_ref, 'version-bump')
+    if: 'github.event_name == ''pull_request'' && contains(github.head_ref, ''version-bump'') && contains(github.event.pull_request.labels, ''version: revision'')'
```
Comment (nicflores) [inline .github/workflows/ci.yml:220]: are the double single quotes needed?

File: core/src/main/scala/precog/SbtPrecogBase.scala (hunk 1 above)
Comment (djspiewak) [reply core/src/main/scala/precog/SbtPrecogBase.scala:259]: Yeah I think it's fine. We were already using `HEAD` implicitly before.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (djspiewak) [reply .github/workflows/ci.yml:220]: Yep, because YAML

**Review by nicflores — APPROVED** [review]

**Pull request merged**